camelot/core/files/storage.py

```diff
@@ -214,11 +214,25 @@ def checkout_stream(self, stored_file: StoredFile) -> BinaryIO:
         self.available()
         return Path(self._path(stored_file.name)).open('rb')
 
-    def delete(self, name: PurePath):
+    def delete(self, name: PurePath, recursive=False):
         """
         :param name: The name of the file to be deleted
+        :param recursive: whether to remove directories and their contents recursively, False by default.
         """
-        Path(self._path(name)).unlink(missing_ok=True)
+        path = Path(self._path(name))
+        if recursive and os.path.isdir(path):
+            self.rmdir(path)
+        else:
+            path.unlink(missing_ok=True)
+
+    def rmdir(self, directory: PurePath):
```

Comment:
di hier wel gevaarlijk, rmdir is denk ik best een 'internal method' die een Path als argument heeft en geen PurePasth

alle andere methods vd storage werken op basis van een relatief path tov de media root, terwijl rmdir op basis van 'een path' werkt, dat kan snel de vergissing tot gevolg hebben dat er een willekeurige folder w verwijderd

Reply:
Ok , fixed in https://github.com/v-finance/camelot/pull/634/commits/ef8e94202a0c7318803fd73e068abf6a42cb7251 & https://github.com/v-finance/camelot/pull/634/commits/8a73753c3df30cbbf32134f84d5287ace6fbbf13.